src/main/java/org/folio/search/service/browse/CallNumberBrowseQueryProvider.java

```diff
@@ -65,7 +66,7 @@ public SearchSourceBuilder get(BrowseRequest request, BrowseContext ctx, boolean
 
   private QueryBuilder getQuery(BrowseContext ctx, String tenantId, int size, boolean isBrowsingForward) {
     var anchor = ctx.getAnchor();
-    var callNumberAsLong = callNumberTermConverter.convert(anchor);
+    var callNumberAsLong = callNumberProcessor.getCallNumberAsLong(anchor);
```

Comment:
the anchor from context is always a shelfKey.
So it's no need to generate from it shelfKey again.